### ISSUE TITLE
Nested 3rd layer @Id template not working on tomcat

### ISSUE THREAD

**Issue by alexberazouski** [vaadin/flow#2438]
Related to #2245 

Can be reproduced on 
https://github.com/vaadin/bakery-app-starter-flow-spring/commit/8bed328adbfdf9bcc36f98638c7576adede9d18a

1. Build: mvn clean install -Dvaadin.productionMode
2. Deploy on tomcat
3. open Users or Product page
4. Server side settings were not applied for search panel (look here: ItemsView.java, UsersView.java, ProductsView.java)

Bug is not reproduced after simple project run : spring-boot:run (either with -Dvaadin.productionMode or without).

On server side proper template object of BakerySearch created. But for some reason it's not connected with client side.

**Comment by johannesh2** [comment]
Thank you. We haven't been able to verify this yet as we haven't been able to get latest alpha to work yet. We'll get back to you.

**Comment by alexberazouski** [comment]
Yes, it still can be reproduced:
https://github.com/vaadin/bakery-app-starter-flow-spring/tree/bug/flow-nested-template

**Comment by caalador** [comment]
Which tomcat?
What parts should I look at to know that they are wrong?

**Comment by alexberazouski** [comment]
All the steps are described in the issue. 

You can see from the latest commit of branch that I've provided what code exactly was changed to reproduce the issue.

**Comment by caalador** [comment]
So any standalone tomcat version or a eclipse handled tomcat?
Could you say in the ticket that what is missing so I don't have to go through the code.

**Comment by alexberazouski** [comment]
![screen shot 2017-10-11 at 15 45 17](https://user-images.githubusercontent.com/28781210/31441261-4bc1a916-ae9b-11e7-82cf-9483064ff3dc.png)

yep, standalone tomcat 8.5.20

**Comment by alexberazouski** [comment]
BakerySearch server-side component settings are not applied to the client-side template. Button click listener; button text, search field behavior. It is reproduced only after war file deployed on any tomcat. And isn't reproduced when app is run via spring-boot

**Comment by caalador** [comment]
I can't get it to build:
```
[ERROR] bower paper-tabs#^2.0.0                 EINVRES Request to https://bower.herokuapp.com/packages/paper-tabs failed with 410
[ERROR] error Command failed with exit code 1.
```

**Comment by alexberazouski** [comment]
We will fix this soon

**Comment by alexberazouski** [comment]
Made fix commit to the current branch

**Comment by denis-anisimov** [comment]
There is a theoretical possibility that it's the same as #2653 (even though it states it's Spring related issue).

**Comment by pleku** [comment]
Putting this on hold until the starter is using the new _official_ Spring add-on with the fix for #2653. 

Ping @johannesh2

**Comment by denis-anisimov** [comment]
#2653 doesn't relate to Spring directly.
The fix has no any relation to Spring at least.

**Comment by denis-anisimov** [comment]
This is the same as #2987

**Comment by denis-anisimov** [comment]
Superseded by #3057
